plutus-core-interpreter/test/CekMachine.hs

```diff
@@ -0,0 +1,52 @@
+-- | The CEK machine tests.
+
+{-# LANGUAGE OverloadedStrings #-}
+module CekMachine
+    ( test_evaluateCek
+    ) where
+
+import           Language.PlutusCore
+import           Language.PlutusCore.Constant
+import           Language.PlutusCore.Evaluation.Result      (evaluationResultToMaybe)
+import           Language.PlutusCore.Generators
+import           Language.PlutusCore.Generators.Interesting
+import           PlutusCoreInterpreter.CekMachine
+
+import           Control.Monad.Morph
+import           Data.Foldable
+import           Hedgehog                                   hiding (Size, Var)
+import           Test.Tasty
+import           Test.Tasty.Hedgehog
+
+-- TODO: also type check the terms.
+propTermOfTypedBuiltinValue :: GenT Quote (TermOf (TypedBuiltinValue Size a)) -> Property
+propTermOfTypedBuiltinValue genTermOfTbv = property . hoist (return . runQuote) $ do
+    TermOf term tbv <- forAllPrettyCfgT genTermOfTbv
+    resExpected <- lift $ unsafeMakeBuiltin tbv
+    for_ (evaluationResultToMaybe $ evaluateCek term) $ \resActual ->
```

Comment:
I'm nitpicking now, but this function is also duplicated and just takes a generic evaluator. Perhaps it could go in the support module?

Reply:
Agreed.